cloudinstall/controllers/installbase.py

```diff
@@ -110,9 +112,8 @@ def update(self, *args, **kwargs):
             self.ui.render_machine_wait_view(self.config)
             self.loop.redraw_screen()
 
-        alarm = self.loop.set_alarm_in(1, self.update)
-        self.observe('stop alarm', partial(self.loop.remove_alarm,
-                                           alarm))
+        AlarmMonitor.add_alarm(self.loop.set_alarm_in(1, self.update))
+        self.observe('stop alarm', AlarmMonitor.abort)
```

Comment:
I think this will work, I don't think we really have any other 'events' that we would be listening for anyway except for a raised exception.

I'll update the code to use that. I'll probably leave the observer though in case down the line we want to listen for any other events.